src/components/ApplicationLayout/ApplicationLayout.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+import type { PropsWithSpread } from "types";
+import classNames from "classnames";
+import type { ReactNode } from "react";
+import { useState, type HTMLProps, type PropsWithChildren } from "react";
+
+import Panel from "components/Panel";
+import type { PanelProps } from "components/Panel";
+import type { PanelLogoDefaultElement } from "components/Panel";
+import type { ExclusiveProps } from "types";
+
+import type { SideNavigationProps } from "../SideNavigation";
+import SideNavigation from "../SideNavigation";
+import type { SideNavigationLinkDefaultElement } from "../SideNavigation/SideNavigationLink/index";
+
+import AppMain from "./AppMain";
+import AppNavigation from "./AppNavigation";
+import AppNavigationBar from "./AppNavigationBar";
+import AppStatus from "./AppStatus";
+import Application from "./Application";
+import Button from "components/Button";
+import Icon from "components/Icon";
+
+export type BaseProps<
+  NI = SideNavigationLinkDefaultElement,
+  PL = PanelLogoDefaultElement,
+> = PropsWithSpread<
+  {
+    aside?: ReactNode;
+    dark?: boolean;
+    logo: NonNullable<PanelProps<PL>["logo"]>;
+    navLinkComponent?: SideNavigationProps<NI>["linkComponent"];
+    menuCollapsed?: boolean;
+    menuPinned?: boolean;
+    navPanelClassName?: string;
+    onCollapseMenu?: (menuCollapsed: boolean) => void;
+    onPinMenu?: (menuPinned: boolean) => void;
+    status?: ReactNode;
+  } & PropsWithChildren,
+  HTMLProps<HTMLDivElement>
+>;
+
+export type Props<
+  NI = SideNavigationLinkDefaultElement,
+  PL = PanelLogoDefaultElement,
+> = BaseProps<NI, PL> &
+  ExclusiveProps<
+    {
+      navItems: NonNullable<SideNavigationProps<NI>["items"]>;
+    },
+    {
+      sideNavigation: ReactNode;
+    }
+  >;
+
+const ApplicationLayout = <
```

Comment:
I am not sure a combination of the other components is generic enough to have it in react-components. I'd assume our different applications have too different requirements to allow a reuse of this combination. Wdyt?

Reply:
This component has been working for the apps that I've tested in so far (ReBAC Admin, Juju Dashboard and Admin UI), so it's possible we can use this for common cases and then some apps might need to have their own implementation.

I plan to keep testing this with other apps to see where it doesn't meet our needs.

Reply:
It looks like MAAS-UI and and LXD-UI have some custom menu features that mean that the `<ApplicationLayout>` component doesn't quite work, or at least not without making it more complicated. In both these cases they can be composed using the child components.

https://github.com/canonical/lxd-ui/pull/768
https://github.com/canonical/maas-ui/pull/5434